Tools/serial/generate_config.py

```diff
@@ -211,6 +224,80 @@ def parse_yaml_parameters_config(yaml_config):
         param_group = parameters_section.get('group', None)
         for param_name in definitions:
             param = definitions[param_name]
+            if 'ethernet' in param:
```

Comment:
Can you avoid the code duplication here, and instead pass `has_ethernet_support`?